packages/swayze/lib/src/core/controller/cells/cells_controller.dart

```diff
@@ -238,9 +243,18 @@ class SwayzeCellsController<CellDataType extends SwayzeCellData>
     // finds the a suitable coordinate.
     do {
       newCoordinate += _moveVectors[direction]!;
+
+      final dx = maxElasticColumns != null
+          ? min(newCoordinate.dx, max(maxElasticColumns - 1, columnsCount - 1))
+          : newCoordinate.dx;
+
+      final dy = maxElasticRows != null
+          ? min(newCoordinate.dy, max(maxElasticRows - 1, rowsCount - 1))
+          : newCoordinate.dy;
```

Comment:
this prevents the user from being able to select cells that exceed the max amount allowed of elastic columns or rows